specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,639 @@
+Applications that use VOA (via a VOA library) can therefore opt to pass a list of _os_ strings, if verifiers are known to be spread over different _os_ directories.
+VOA users can omit specific parts, e.g. `IMAGE_ID` and `IMAGE_VERSION`, if they know that these parts are not used in the [VOA hierarchy] for their _os_.
+
+### Purpose
+
+A _purpose_ combines a _role_ and a usage _mode_:
+
+- A _role_ acts as a trust domain (e.g. the "package" _role_ is used for signatures for package verification).
+- There are two _modes_ in which _signature verifiers_ can be stored in VOA:
+  - Verifiers used for direct artifact verification
+  - Verifiers that serve as _trust anchors_
+
+A _role_ and a _mode_ in combination form a _purpose_.
+
+_Trust anchor_ verifiers are always used to ascertain the validity of the associated _artifact verifiers_ (e.g. _trust anchor_ verifiers for the "package" _role_ are used to validate the _artifact verifiers_ for the "package" _role_).
+
+To use _signature verifiers_ in more than one _mode_ [symlinking] may be used.
+
+#### Directory naming
+
+_Purpose_ directories for direct _artifact verifiers_ are named `$role` (e.g. `package`), while _trust anchors_ are stored below directories named `trust-anchor-$role` (e.g. `trust-anchor-package`).
+
+Note that _trust anchor_ directory names always start with a "trust-anchor-" fragment.
+
+Either two or one _purpose_ directories may exist per _role_:
+
+- One directory which contains _artifact verifiers_ (e.g. `package`), and a second directory which contains the corresponding _trust anchors_ (e.g. `trust-anchor-package`).
+- Just one directory which contains _artifact verifiers_ (e.g. `package`).
+
+##### Two purpose directories: Verification relying on trust anchors
+
+For example, verifiers for the "package" _role_ may be stored in two _purpose_ directories:
+
+- `package` ("_artifact verifiers_ of actors who are designated for package signing"), and
+- `trust-anchor-package` ("_trust anchors_ for _artifact verifiers_ used for package signing")
+
+In this scenario, verification based on _trust anchors_ is performed.
+
+##### One purpose directory: Direct verification
+
+In another example, verifiers for the "image" _role_ may be stored in only one _purpose_ directory:
+
+- `image` ("_artifact verifiers_ of actors who are designated for OS image signing")
+
+In this scenario, only direct verification can be performed.
+
+#### Roles as trust domains
+
+Having distinct _roles_ allows the use of separate sets of _signature verifiers_ per role.
+This is useful if different actors are expected to issue signatures for each _role_.
+Thus, each _role_ acts as a trust domain, e.g. the "package" _role_ is used for signatures for packages, while the "image" _role_ is used for signatures for OS images.
+The work on each of the _roles_ may be performed by different teams, using different verifiers.
+
+#### Default roles
+
+The default roles are:
```

Comment:
do you mean "default"? 

don't you want to say: "Standard roles defined by this spec"?